core/app/models/spree/product.rb

```diff
@@ -105,7 +105,7 @@ class Product < Spree::Base
       validates :price, if: proc { Spree::Config[:require_master_price] }
     end
 
-    validates :slug, length: { minimum: 3 }, allow_blank: true, uniqueness: true
+    validates :slug, allow_blank: true, uniqueness: true
```

Comment:
Just tested this out:

1)

```ruby
validates :slug, uniqueness: true, presence: true
```

vs

2)

```ruby
validates :slug, uniqueness: { allow_blank: true }, presence: true
```

1) `validates :slug, uniqueness: true, presence: true`
(emphasis mine)

>[1] pry(main)> p=Spree::Product.last
  Spree::Product Load (0.7ms)  SELECT  "spree_products".* FROM "spree_products" WHERE "spree_products"."deleted_at" IS NULL ORDER BY "spree_products"."id" DESC LIMIT $1  [["LIMIT", 1]]
=> #<Spree::Product:0x007faa674a0420
 id: 2,
 name: "Asd product",
 description: nil,
 available_on: nil,
 deleted_at: nil,
 slug: "asd-product-51ed984b-f177-4b30-8c54-e2ce98286349",
 meta_description: nil,
 meta_keywords: nil,
 tax_category_id: nil,
 shipping_category_id: 1,
 created_at: Wed, 31 May 2017 09:20:58 UTC +00:00,
 updated_at: Wed, 31 May 2017 09:20:58 UTC +00:00,
 promotionable: true,
 meta_title: nil,
 discontinue_on: nil>
[2] pry(main)> p.slug=''
=> ""
[3] pry(main)> p.valid?
  Spree::Variant Load (0.6ms)  SELECT  "spree_variants".* FROM "spree_variants" WHERE "spree_variants"."deleted_at" IS NULL AND "spree_variants"."product_id" = $1 AND "spree_variants"."is_master" = $2 LIMIT $3  [["product_id", 2], ["is_master", "t"], ["LIMIT", 1]]
  Spree::Preference Load (0.5ms)  SELECT  "spree_preferences".* FROM "spree_preferences" WHERE "spree_preferences"."key" = $1 LIMIT $2  [["key", "spree/app_configuration/currency"], ["LIMIT", 1]]
  Spree::Price Load (0.6ms)  SELECT  "spree_prices".* FROM "spree_prices" WHERE "spree_prices"."variant_id" = $1 AND "spree_prices"."currency" = $2 LIMIT $3  [["variant_id", 2], ["currency", "USD"], ["LIMIT", 1]]
  Spree::ShippingCategory Load (0.5ms)  SELECT  "spree_shipping_categories".* FROM "spree_shipping_categories" WHERE "spree_shipping_categories"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  Spree::Preference Load (0.3ms)  SELECT  "spree_preferences".* FROM "spree_preferences" WHERE "spree_preferences"."key" = $1 LIMIT $2  [["key", "spree/app_configuration/require_master_price"], ["LIMIT", 1]]
  **Spree::Product Exists (1.4ms)  SELECT  1 AS one FROM "spree_products" WHERE "spree_products"."slug" = $1 AND "spree_products"."deleted_at" IS NULL AND ("spree_products"."id" != $2) LIMIT $3  [["slug", ""], ["id", 2], ["LIMIT", 1]]**
=> false
[4] pry(main)> p.slug=nil
=> nil
[5] pry(main)> p.valid?
  Spree::Product Exists (0.8ms)  SELECT  1 AS one FROM "spree_products" INNER JOIN "friendly_id_slugs" ON "friendly_id_slugs"."sluggable_id" = "spree_products"."id" AND "friendly_id_slugs"."deleted_at" IS NULL AND "friendly_id_slugs"."sluggable_type" = $1 WHERE ("spree_products"."id" != 2) AND "friendly_id_slugs"."deleted_at" IS NULL AND (sluggable_id <> 2) AND "spree_products"."slug" = 'asd-product' LIMIT $2  [["sluggable_type", "Spree::Product"], ["LIMIT", 1]]
  Spree::Product Exists (0.5ms)  SELECT  1 AS one FROM "spree_products" INNER JOIN "friendly_id_slugs" ON "friendly_id_slugs"."sluggable_id" = "spree_products"."id" AND "friendly_id_slugs"."deleted_at" IS NULL AND "friendly_id_slugs"."sluggable_type" = $1 WHERE ("spree_products"."id" != 2) AND "friendly_id_slugs"."deleted_at" IS NULL AND (sluggable_id <> 2) AND "spree_products"."slug" = 'asd-product' LIMIT $2  [["sluggable_type", "Spree::Product"], ["LIMIT", 1]]
  Spree::Preference Load (0.3ms)  SELECT  "spree_preferences".* FROM "spree_preferences" WHERE "spree_preferences"."key" = $1 LIMIT $2  [["key", "spree/app_configuration/require_master_price"], ["LIMIT", 1]]
  Spree::Product Exists (0.3ms)  SELECT  1 AS one FROM "spree_products" WHERE "spree_products"."slug" = $1 AND "spree_products"."deleted_at" IS NULL AND ("spree_products"."id" != $2) LIMIT $3  [["slug", "asd-product-7ff88b91-6b6a-4e7c-af41-70b7cdc5fd09"], ["id", 2], ["LIMIT", 1]]
=> true
[6] pry(main)> p.slug?
=> true
[7] pry(main)> p.slug
=> "asd-product-7ff88b91-6b6a-4e7c-af41-70b7cdc5fd09"


2) `validates :slug, uniqueness: { allow_blank: true }, presence: true`

[1] pry(main)> p=Spree::Product.last
  Spree::Product Load (0.7ms)  SELECT  "spree_products".* FROM "spree_products" WHERE "spree_products"."deleted_at" IS NULL ORDER BY "spree_products"."id" DESC LIMIT $1  [["LIMIT", 1]]
=> #<Spree::Product:0x007f89bd57ad20
 id: 2,
 name: "Asd product",
 description: nil,
 available_on: nil,
 deleted_at: nil,
 slug: "asd-product-51ed984b-f177-4b30-8c54-e2ce98286349",
 meta_description: nil,
 meta_keywords: nil,
 tax_category_id: nil,
 shipping_category_id: 1,
 created_at: Wed, 31 May 2017 09:20:58 UTC +00:00,
 updated_at: Wed, 31 May 2017 09:20:58 UTC +00:00,
 promotionable: true,
 meta_title: nil,
 discontinue_on: nil>
[2] pry(main)> p.slug=''
=> ""
[3] pry(main)> p.valid?
  Spree::Variant Load (0.8ms)  SELECT  "spree_variants".* FROM "spree_variants" WHERE "spree_variants"."deleted_at" IS NULL AND "spree_variants"."product_id" = $1 AND "spree_variants"."is_master" = $2 LIMIT $3  [["product_id", 2], ["is_master", "t"], ["LIMIT", 1]]
  Spree::Preference Load (0.3ms)  SELECT  "spree_preferences".* FROM "spree_preferences" WHERE "spree_preferences"."key" = $1 LIMIT $2  [["key", "spree/app_configuration/currency"], ["LIMIT", 1]]
  Spree::Price Load (0.4ms)  SELECT  "spree_prices".* FROM "spree_prices" WHERE "spree_prices"."variant_id" = $1 AND "spree_prices"."currency" = $2 LIMIT $3  [["variant_id", 2], ["currency", "USD"], ["LIMIT", 1]]
  Spree::ShippingCategory Load (0.6ms)  SELECT  "spree_shipping_categories".* FROM "spree_shipping_categories" WHERE "spree_shipping_categories"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  Spree::Preference Load (0.3ms)  SELECT  "spree_preferences".* FROM "spree_preferences" WHERE "spree_preferences"."key" = $1 LIMIT $2  [["key", "spree/app_configuration/require_master_price"], ["LIMIT", 1]]
=> false
[4] pry(main)> p.slug=nil
=> nil
[5] pry(main)> p.valid?
  Spree::Product Exists (0.9ms)  SELECT  1 AS one FROM "spree_products" INNER JOIN "friendly_id_slugs" ON "friendly_id_slugs"."sluggable_id" = "spree_products"."id" AND "friendly_id_slugs"."deleted_at" IS NULL AND "friendly_id_slugs"."sluggable_type" = $1 WHERE ("spree_products"."id" != 2) AND "friendly_id_slugs"."deleted_at" IS NULL AND (sluggable_id <> 2) AND "spree_products"."slug" = 'asd-product' LIMIT $2  [["sluggable_type", "Spree::Product"], ["LIMIT", 1]]
  Spree::Product Exists (0.5ms)  SELECT  1 AS one FROM "spree_products" INNER JOIN "friendly_id_slugs" ON "friendly_id_slugs"."sluggable_id" = "spree_products"."id" AND "friendly_id_slugs"."deleted_at" IS NULL AND "friendly_id_slugs"."sluggable_type" = $1 WHERE ("spree_products"."id" != 2) AND "friendly_id_slugs"."deleted_at" IS NULL AND (sluggable_id <> 2) AND "spree_products"."slug" = 'asd-product' LIMIT $2  [["sluggable_type", "Spree::Product"], ["LIMIT", 1]]
  Spree::Preference Load (0.4ms)  SELECT  "spree_preferences".* FROM "spree_preferences" WHERE "spree_preferences"."key" = $1 LIMIT $2  [["key", "spree/app_configuration/require_master_price"], ["LIMIT", 1]]
  Spree::Product Exists (0.5ms)  SELECT  1 AS one FROM "spree_products" WHERE "spree_products"."slug" = $1 AND "spree_products"."deleted_at" IS NULL AND ("spree_products"."id" != $2) LIMIT $3  [["slug", "asd-product-2b4bed52-afaa-4348-b527-a156df067ad0"], ["id", 2], ["LIMIT", 1]]
=> true
[6] pry(main)> p.slug
=> "asd-product-2b4bed52-afaa-4348-b527-a156df067ad0"